catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SetVariableTest.java

```diff
@@ -0,0 +1,170 @@
+		onView(withText(R.string.deletion_alert_yes)).perform(click());
+		pressBack();
+		pressBack();
+
+		onView(withText(userVariableName)).check(doesNotExist());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(userVariableName))
+				.check(doesNotExist());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(secondUserVariableName))
+				.check(matches(isDisplayed()));
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateUserVariableInFormulaEditor() {
+		String userVariableName = "testVariable1";
+		onBrickAtPosition(1).onChildView(withId(R.id.brick_set_variable_edit_text)).perform(click());
+		onView(withId(R.id.formula_editor_keyboard_data)).perform(click());
+		onView(withId(R.id.button_add)).perform(click());
+		onView(withId(R.id.dialog_formula_editor_data_name_edit_text)).perform(typeText(userVariableName));
+		onView(withText(R.string.ok)).perform(click());
+		onView(allOf(withChild(withText(userVariableName)), withChild(withText("0.0")))).check(matches(isDisplayed()));
+		pressBack();
+		onView(allOf(withParent(withId(R.id.set_variable_spinner)), withText(userVariableName))).check(matches
+				(isDisplayed()));
+		pressBack();
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(userVariableName))
+				.check(matches(isDisplayed()));
+		onBrickAtPosition(1).checkShowsText(userVariableName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testViewInFormulaEditorAfterClone() {
+		//There is no clonening in the original test, why this name?
+		String userVariableName = "testVariable1";
+		String userVariableNameTwo = "testVariable2";
+		onBrickAtPosition(1).onChildView(withId(R.id.brick_set_variable_edit_text)).perform(click());
+		onView(withId(R.id.formula_editor_keyboard_data)).perform(click());
+		onView(withId(R.id.button_add)).perform(click());
+		onView(withId(R.id.dialog_formula_editor_data_name_edit_text)).perform(typeText(userVariableName));
+		onView(withText(R.string.ok)).perform(click());
+		pressBack();
+		onView(withId(R.id.formula_editor_keyboard_ok)).perform(click());
+		UiTestUtils.addBrick(SetVariableBrick.class, R.string.category_data);
+		// manual perform new variable because of CAT-2359
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).perform(click());
+		onView(withText(R.string.brick_variable_spinner_create_new_variable))
+				.perform(click());
+		onView(withId(R.id.dialog_formula_editor_data_name_edit_text)).perform(typeText(userVariableNameTwo));
+		onView(withId(android.R.id.button1)).perform(click());
+
+		/* checks if both variables are inside the spinner, to make sure it is the spinner we require it to have also
+		 the create new variable text and require it to display all of them
+		  */
+		onBrickAtPosition(2).onVariableSpinner(R.id.set_variable_spinner).perform(click());
```

Comment:
please add this to the BrickSpinnerDataInteractionWrapper, and then use it from there. (simliar to checkValuesAvailable, but taking strings instead of resource ids)